pubnub-gson/src/main/java/com/pubnub/internal/PubNubImpl.java

```diff
@@ -0,0 +1,627 @@
+     *
+     * @param inputString String to be encrypted
+     * @return String containing the encryption of inputString using cipherKey
+     */
+    @Override
+    public @NotNull String decrypt(@NotNull String inputString) throws PubNubException {
+        return decrypt(inputString, null);
+    }
+
+    /**
+     * Perform Cryptographic decryption of an input string using the cipher key
+     *
+     * @param inputString String to be encrypted
+     * @param cipherKey   cipher key to be used for encryption
+     * @return String containing the encryption of inputString using cipherKey
+     * @throws PubNubException throws exception in case of failed encryption
+     */
+    @Override
+    public @NotNull String decrypt(@NotNull String inputString, String cipherKey) throws PubNubException {
+        if (inputString == null) {
+            throw new PubNubException(PubNubError.INVALID_ARGUMENTS);
+        }
+        return getPubNubCore().decrypt(inputString, cipherKey);
+    }
+
+    @Override
+    public @NotNull InputStream decryptInputStream(@NotNull InputStream inputStream) throws PubNubException {
+        return decryptInputStream(inputStream, null);
+    }
+
+    @Override
+    public @NotNull InputStream decryptInputStream(@NotNull InputStream inputStream, @Nullable String cipherKey) throws PubNubException {
+        if (inputStream == null) {
+            throw new PubNubException(PubNubError.INVALID_ARGUMENTS);
+        }
+        return getPubNubCore().decryptInputStream(inputStream, cipherKey);
+    }
+
+    /**
+     * Perform Cryptographic encryption of an input string and the cipher key provided by PNConfiguration
+     *
+     * @param inputString String to be encrypted
+     * @return String containing the encryption of inputString using cipherKey
+     */
+    @Override
+    public @NotNull String encrypt(@NotNull String inputString) throws PubNubException {
+        return encrypt(inputString, null);
+    }
+
+    /**
+     * Perform Cryptographic encryption of an input string and the cipher key.
+     *
+     * @param inputString String to be encrypted
+     * @param cipherKey   cipher key to be used for encryption
+     * @return String containing the encryption of inputString using cipherKey
+     * @throws PubNubException throws exception in case of failed encryption
+     */
+    @Override
+    public @NotNull String encrypt(@NotNull String inputString, String cipherKey) throws PubNubException {
+        if (inputString == null) {
+            throw new PubNubException(PubNubError.INVALID_ARGUMENTS);
+        }
+        return getPubNubCore().encrypt(inputString, cipherKey);
+    }
+
+    @Override
+    public @NotNull InputStream encryptInputStream(@NotNull InputStream inputStream) throws PubNubException {
+        return encryptInputStream(inputStream, null);
+    }
+
+    @Override
+    public @NotNull InputStream encryptInputStream(@NotNull InputStream inputStream, String cipherKey) throws PubNubException {
```

Comment:
There is a discrepancy. Here and in few methods above there is:
`public @NotNull`
and in other places there is
```
@NotNull
public
```

Reply:
done